devito/operator.py

```diff
@@ -151,14 +150,22 @@ def arguments(self, **kwargs):
         for i in self.parameters:
             if i.is_TensorArgument:
                 assert(i.verify(kwargs.pop(i.name, None)))
-        runtime_dimensions = [d for d in self.dimensions if d.value is not None]
+        runtime_dimensions = [d for d in self.dimensions if not d.is_Fixed]
         for d in runtime_dimensions:
             d.verify(kwargs.pop(d.name, None))
         for i in self.parameters:
             if i.is_ScalarArgument:
                 i.verify(kwargs.pop(i.name, None))
-
-        dim_sizes = OrderedDict([(d.name, d.value) for d in runtime_dimensions])
+        dim_sizes = {}
+        for d in runtime_dimensions:
+            if d.value is not None:
+                _, d_start, d_end = d.value
+                # The variable is called size but what it expects is extent
+                # Since this is used for loop blocking
```

Comment:
I don't like the fact that ``Operator.arguments`` does anything specific to "loop blocking". Can we abstract things a little bit ?

Reply:
It is not specifically for loop blocking. It is simply calculating the extent of the dimension - which is a general feature.